src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -164,7 +164,7 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
                     diagnostics.Add(ErrorCode.ERR_ConstantExpected, patternExpression.Location);
                     hasErrors = true;
                 }
-                else if (inputType.IsPointerType() == true && Compilation.LanguageVersion < MessageID.IDS_FeatureRecursivePatterns.RequiredVersion())
+                else if (inputType.IsPointerOrFunctionPointer() == true && Compilation.LanguageVersion < MessageID.IDS_FeatureRecursivePatterns.RequiredVersion())
                 {
                     // before C# 8 we did not permit `pointer is null`
                     diagnostics.Add(ErrorCode.ERR_PointerTypeInPatternMatching, patternExpression.Location);
```

Comment:
>ERR_PointerTypeInPatternMatching [](start = 46, length = 32)

Not related to this PR, but it looks like the error that we report makes no indication that a language version is involved and IDE is unlikely to offer to bump it. Consider opening an issue.

Reply:
Filed [https://github.com/dotnet/roslyn/issues/43319](https://github.com/dotnet/roslyn/issues/43319).

---
In reply to: [407721019](https://github.com/dotnet/roslyn/pull/43248#discussion_r407721019) [](ancestors = 407721019)